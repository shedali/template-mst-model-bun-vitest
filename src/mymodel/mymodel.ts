import { types as t } from "mobx-state-tree";
export const DistrictModel = t.model("District", {
  id: t.identifier,
  name: t.string,
});
