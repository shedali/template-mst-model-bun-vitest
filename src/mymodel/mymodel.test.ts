import { describe, expect, it } from "vitest";
import { DistrictModel as MyModel } from "./mymodel";
describe("my model", () => {
  it("should be defined", () => {
    expect(MyModel).toBeDefined();
  });
  it("should match snapshot", () => {
    expect(
      MyModel.create({
        id: "1",
        name: "my 1",
      })
    ).toMatchSnapshot();
  });
});
