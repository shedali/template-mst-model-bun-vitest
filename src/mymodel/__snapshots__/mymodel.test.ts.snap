// Vitest Snapshot v1

exports[`my model > should match snapshot 1`] = `
{
  "id": "1",
  "name": "my 1",
}
`;
